storage/src/backend/registry.rs

```diff
@@ -487,8 +491,31 @@ impl RegistryReader {
                 return self._try_read(buf, offset, false);
             }
         } else {
-            resp =
-                self.request::<&[u8]>(Method::GET, url.as_str(), None, headers.clone(), false)?;
+            // resp =
+            //     self.request::<&[u8]>(Method::GET, url.as_str(), None, headers.clone(), false)?;
+            resp = match self.request::<&[u8]>(
+                Method::GET,
+                url.as_str(),
+                None,
+                headers.clone(),
+                false,
+            ) {
+                Ok(res) => res,
+                Err(e) => {
+                    if &url[0..5] != "https" {
```

Comment:
Why not check `self.state.scheme` directly here.

Reply:
`self.state.scheme` has changed, but the `url` has not changed, in some cases

Reply:
Could you point out where the state has changed?